### DISCUSSION TITLE
Workspace folder root keeps changing

### DISCUSSION THREAD

**Discussion by axelson** [emacs-lsp/lsp-mode#3899] [Q&A]
A few times a day I'll notice that all of my lsp servers are broken because I keep getting one of a few things happening:
- The workspace root is now `/home/<myuser>`
- The workspace root is now `<gitrepo>/` when I previously configured it to `<gitrepo>/subdir`

I've been working around this by running `lsp-workspace-folders-remove ` when I notice it and then killing the currently open buffers and re-initializing the buffers which will then re-prompt for the workspace root to use.

Is there something that I can do to make my selection sticky? Or blacklist certain folders?

**Comment by axelson** [comment]
This seems to happen when I'm switching workspaces (since the server for the old workspace shutdowns and the server for the new workspace starts up)

**Reply by yyoncho** [reply]
This sounds like the expected behavior if in both workspaces you have different project roots.

**Reply by axelson** [reply]
Thanks for the response but from my point of view this is not expected behavior. Maybe these reproduction steps will make the behavior I want to avoid more obvious.

Open workspace 1
- Run `lsp-describe-session`
- Prints: `/home/jason/dev/pomodoro`

Open workspace 2
- Prints: `/home/jason/dev/scenic-side-screen/fw`

Switch back to workspace 1
- Run `lsp-describe-session`
- Prints: `/home/jason/dev/pomodoro`

So far so good

Switch back to workspace 2
- Run `lsp-describe-session`:
- Prints `/home/jason/dev/scenic-side-screen`

In that last step the lsp directory has changed. I want it to stay as `/home/jason/dev/scenic-side-screen/fw`

Also maybe this screen recording will help illuminate the issues I'm having (it includes the steps above, but also includes manually removing `/home/jason` which is getting set as a workspace root automatically somehow:

https://user-images.githubusercontent.com/9973/212483691-ea592cb0-b08c-4476-a526-872f9b40b783.mp4

Note: if I restart emacs then I'll have to follow the exact same steps again, it's almost like my workspace folder selections are not being updated.

**Reply by yyoncho** [reply]
what is your lsp-auto-guess-root?

**Reply by axelson** [reply]
`lsp-auto-guess-root` is `nil`

**Reply by axelson** [reply]
@yyoncho Aha! I just found the culprit. I only get this behavior when I am using `(lsp-treemacs-sync-mode 1)`
